README.md

```diff
@@ -108,3 +124,23 @@ err := publisher.Publish(
     pubsub.SetMessageDelay(5*time.Second),
 )
 ```
+
+### Publishing a message with a message ID
+The duplication detection feature requires messages to have a messageID, as messageID is the key ServiceBus will de-dupe on.
+Refer to the [Initializing a publisher with duplication detection section](#initializing-a-publisher-with-duplication-detection).
+```
+cmd := &ClusterStatusChanged{
+    Event: Event{Id: uuidVal},
+    Command: Command{
+        Id:            uuid.New(),
+        DestinationId: "destination",
+    },
+}
+messageID := "someMessageIDWithBusinessContext"
+err := publisher.Publish(
+    ctx,
+    cmd,
+    pubsub.SetMessageID(messageID),
```

Comment:
so, the messageID is set as part of the Publish, and not on the event struct itself.
I think I'm fine with that.
what do you think of saying that if the message has a property MessageID, we use that by convention

Reply:
keep the Publish func option param, but add a convention too

Reply:
separate PR, obviously :)

Reply:
i thought about that actually, but i thought parsing things underneath will cause more confusion and potential bugs. I think forcing the publisher to explicitly set message ID is clearer